src/plugins/headers.js

```diff
@@ -0,0 +1,23 @@
+export const headers = {
+  plugin: {
+    name: 'headers',
+    register: (server, _options) => {
+      server.ext('onPreResponse', (request, h) => {
+        const response = request.response
+
+        const headers = response.headers || response.output?.headers
```

Comment:
If normal response then headers go on `response.headers`, if error, then it will be `response.output.headers`